crates/ty_python_semantic/src/types.rs

```diff
@@ -5708,9 +5708,7 @@ impl<'db> Type<'db> {
                             ],
                         });
                     };
-                    let instance = Type::ClassLiteral(class).to_instance(db).expect(
-                        "nearest_enclosing_class must return type that can be instantiated",
-                    );
```

Comment:
This would previously use the "default" specialization, and so we would end up with an upper bound of `Container[bytes]` for the example in the mdtest. With the change here, we instead use `Container[Unknown]` as the upper bound.